### PR TITLE
persist image overrides for Editions

### PR DESCRIPTION
**Pull request by akash1810**
## What's changed?
[Card](https://trello.com/c/a2P58r1x/36-edit-the-relevant-metadata-fields-for-editions-exact-fields-to-store-tbc)

Stores image overrides.

Also splits out the database model from the client model with the intention of the database model being a simpler representation that gets serialised into the client model.

## Implementation notes
n/a

### General
- [x] 🤖 Relevant tests added
- [x] ✅ CI checks / tests run locally
- [ ] 🔍 Checked on CODE

### Client
- [x] 🚫 No obvious console errors on the client (i.e. React dev mode errors)
- [x] 🎛️ No regressions with existing user interactions (i.e. all existing buttons, inputs etc. work)
- [ ] 📷 Screenshots / GIFs of relevant UI changes included

### CODE REVIEW

File: app/model/editions/EditionsClientCollection.scala [hunk 1]
```diff
@@ -1,57 +1,58 @@
 package model.editions
 
+import model.editions.client.ClientArticleMetadata
 import play.api.libs.json.Json
 
 // Ideally the frontend can be changed so we don't have this weird modelling!
 
-case class EditionsFrontendArticle(id: String, frontPublicationDate: Long, meta: Option[ArticleMetadata])
+case class EditionsClientArticle(id: String, frontPublicationDate: Long, meta: Option[ClientArticleMetadata])
```
Comment (jonathonherbert) [inline app/model/editions/EditionsClientCollection.scala]: I really like the naming here, 👍

File: app/model/editions/client/ClientArticleMetadata.scala [hunk 2]
```diff
@@ -0,0 +1,99 @@
+package model.editions.client
+
+import model.editions.{ArticleMetadata, Image, ImageOption}
+import play.api.libs.json.Json
+
+case class ClientArticleMetadata (
```
Comment (jonathonherbert) [inline app/model/editions/client/ClientArticleMetadata.scala]: Huh -- how did we get away with not modelling this before? Did we just transparently pass the overrides through as JSON?

Comment (akash1810) [reply app/model/editions/client/ClientArticleMetadata.scala]: Looks like we have some models [here](https://github.com/guardian/facia-scala-client/blob/master/facia-json/src/main/scala/com/gu/facia/client/models/Collection.scala#L126) which we could reuse.

**Comment by akash1810** [comment]
For some reason, `imageCutoutReplace` is always `true` which isn't correct. DNM for now.

File: app/model/editions/ImageOption.scala [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import enumeratum.{EnumEntry, PlayEnum}
+import enumeratum.EnumEntry.Camelcase
+
+sealed abstract class ImageOption extends EnumEntry with Camelcase
+object ImageOption extends PlayEnum[ImageOption] {
+  case object UseArticleTrail extends ImageOption
+  case object Hide extends ImageOption
+  case object ReplaceCutout extends ImageOption
+  case object ReplaceSlideshow extends ImageOption
```
Comment (sihil) [inline app/model/editions/ImageOption.scala]: I'm 99% sure that we're not supporting slideshows but @blishen should be able to clarify.

Comment (sihil) [reply app/model/editions/ImageOption.scala]: I understand this came from @blishen so all good - just sounds a little like scope creep 😿

**Comment by akash1810** [comment]
I'm going to split this out into multiple PRs tomorrow. One for image stuff (which currently has a client side bug as far as I can tell) and one for the rest. This'll mean we deliver value faster. I'll also write tests against the image stuff for my sanity.

**Comment by sihil** [comment]
I've merged the first couple of commits in #844 and also rebased this on top of master. You might want to hard reset against origin.

File: app/model/editions/EditionsArticle.scala [hunk 4]
```diff
@@ -1,35 +1,72 @@
 package model.editions
 
-import com.gu.editions.{PublishedArticle, PublishedFurniture}
+import com.gu.editions.{PublishedArticle, PublishedFurniture, MediaUrl}
 import play.api.libs.json.Json
 import scalikejdbc.WrappedResultSet
 
-case class ArticleMetadata(
+case class Image (
```
Comment (akash1810) [inline app/model/editions/EditionsArticle.scala]: This and `MediaUrl` would be the same model - need to confirm with @AWare about the structure needed.

File: app/model/editions/client/ClientArticleMetadata.scala [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      articleMetadata.headline,
+      articleMetadata.customKicker,
+      articleMetadata.customKicker.map(_ => true),
+      articleMetadata.trailText,
+      articleMetadata.showQuotedHeadline,
+      articleMetadata.showByline,
+      articleMetadata.byline,
+
+      articleMetadata.imageOption match {
+        case Some(io) if io == ImageOption.Hide => Some(true)
```
Comment (akash1810) [inline app/model/editions/client/ClientArticleMetadata.scala]: Must be a better way to do this! `if articleMetadata.imageOption is defined and == ImageOption.Hide then Some(true) else None`

Comment (sihil) [reply app/model/editions/client/ClientArticleMetadata.scala]: I think you want `articleMetadata.imageOption.collect { case ImageOption.Hide => true }`

Comment (sihil) [reply app/model/editions/client/ClientArticleMetadata.scala]: You can use this in other places where your default pattern maps to `None`

**Pull request merged**